spec/lib/cyclonedx/report_ruby_gems_cyclonedx_spec.rb

```diff
@@ -0,0 +1,155 @@
+            "version": "",
+            "purl": "pkg:gem/master_lock",
+            "properties": [
+              {
+                "key": "source",
+                "value": "git@github.com:coinbase/master_lock.git"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Gemfile"
+              }
+            ]
+          }
+        ]
+      )
+    end
+
+    it 'should report all deps in Gemfile.lock in cyclonedx' do
+      repo = Salus::Repo.new('spec/fixtures/report_ruby_gems/lockfile')
+      scanner = Salus::Scanners::ReportRubyGems.new(repository: repo, config: {})
+      scanner.run
+
+      ruby_cyclonedx = Cyclonedx::ReportRubyGems.new(scanner.report)
+      expected = [
+        {
+          "bom-ref": "pkg:gem/actioncable@5.1.2",
+          "type": "library",
+          "group": "",
+          "name": "actioncable",
+          "version": "5.1.2",
+          "purl": "pkg:gem/actioncable@5.1.2",
+          "properties": [
+            {
+              "key": "source",
+              "value": "rubygems repository https://rubygems.org/ or installed locally"
+            },
+            {
+              "key": "dependency_file",
+              "value": "Gemfile.lock"
+            }
+          ]
+        },
+        {
+          "bom-ref": "pkg:gem/actionmailer@5.1.2",
+          "type": "library",
+          "group": "",
+          "name": "actionmailer",
+          "version": "5.1.2",
+          "purl": "pkg:gem/actionmailer@5.1.2",
+          "properties": [
+            {
+              "key": "source",
+              "value": "rubygems repository https://rubygems.org/ or installed locally"
+            },
+            {
+              "key": "dependency_file",
+              "value": "Gemfile.lock"
+            }
+          ]
+        },
+        {
+          "bom-ref": "pkg:gem/kibana_url@1.0.1",
+          "type": "library",
+          "group": "",
+          "name": "kibana_url",
+          "version": "1.0.1",
+          "purl": "pkg:gem/kibana_url@1.0.1",
+          "properties": [
+            {
+              "key": "source",
+              "value": "rubygems repository https://rubygems.org/ or installed locally"
+            },
+            {
+              "key": "dependency_file",
+              "value": "Gemfile.lock"
+            }
+          ]
+        },
+        {
+          "bom-ref": "pkg:gem/master_lock@0.9.1",
+          "type": "library",
+          "group": "",
+          "name": "master_lock",
+          "version": "0.9.1",
+          "purl": "pkg:gem/master_lock@0.9.1",
+          "properties": [
+            {
+              "key": "source",
+              "value": "git@github.com:coinbase/master_lock.git"
+            },
+            {
+              "key": "dependency_file",
+              "value": "Gemfile.lock"
+            }
+          ]
+        }
+      ]
+      expect(ruby_cyclonedx.build_components_object).to include(*expected)
```

Comment:
This is part of the Gemfile.lock
`
  specs:
    actioncable (5.1.2)
      actionpack (= 5.1.2)
      nio4r (~> 2.0)
  ...
`
Are nested deps like `actionpack`, `nio4r` included in the output too?  If so, could you add one of these?

Reply:
Yep they are included 👍 `actioncable` itself is a nested dependency of rails. Happy to add in some more examples to the spec.